okio/src/commonMain/kotlin/okio/FileHandle.kt

```diff
@@ -64,7 +66,7 @@ abstract class FileHandle(
     arrayOffset: Int,
     byteCount: Int
   ): Int {
-    synchronized(this) {
+    synchronized(aLock) {
```

Comment:
TODO revert based on docs found by @swankjesse 

> There is no need to replace synchronized blocks and methods that are used infrequently (e.g., only performed at startup) or that guard in-memory operations. As always, strive to keep locking policies simple and clear.

Reply:
Nah, this is fine.

Reply:
It's still an extra field (no allocation) in each FileHandle. Your call, but I don't mind putting back synchronized, just for this case.